js/ad_manager_vast.js

```diff
@@ -784,11 +972,17 @@ OO.Ads.manager(function(_, $) {
      * @param {boolean} failed If true, the ending of this ad was caused by a failure
      */
     var _endAd = _.bind(function(ad, failed) {
-      if (ad) {
-        if (ad.isLinear) {
+      if (currentAd && ad) {
+        var isLinear = ad.vpaidAd ? ad.vpaidAd.getAdLinear() : false;
+        if (ad.isLinear || isLinear) {
```

Comment:
Can we move the ad.isLinear check into the calculation for isLinear? This way we can have a check against only isLinear.